codegen/facelift/facelift-codegen.py

```diff
@@ -363,10 +369,17 @@ def run_generation(input, output, dependency, libraryName, all):
             log.debug('process module %s' % module.module_name)
             ctx.update({'path': module_path})
             generateFile(generator, 'module/{{path}}/ModulePrivate.h', 'ModulePrivate.template.h', ctx, libraryName, "")
+            if not module.structs:
+                addToNoMocList('module/{}/ModulePrivate.h'.format(module_path), output, noMocPath)
+
```

Comment:
In fact you open noMocPath multiple times. You could optimize it out to open once and pass a handle instead of path. This should be a bit faster (I don't know if this optimization is worth something).

Reply:
agree.
it's not noticeable now, but if there will be e.g. 4000 line insertion it is very slow (I checket it out).
So yep - changed. The file opens only once and closes after leaving the scope.